nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -161,9 +160,6 @@ in {
     })
     (mkIf cfg.nginx.enable {
       assertions = [
-        { assertion = nginx.enable;
```

Comment:
how is that unreachable? That if was for services.cryptpad.nginx.enable, and this check for services.nginx.enable

. . .anyway, you force enabled services.nginx in a later commit, so I guess now it really is unreachable.
Okay. I think like ACME this probably ought to be a conscious decision but I don't mind either way

Reply:
Sorry, I've been working on this while you were rebasing so this may still be a bit outdated. I think this gets solved by making the entire Nginx reverse proxy optional like so: https://github.com/NixOS/nixpkgs/commit/73e025a03b84383b4e873ec68bf72bbbcf990c9d

When I'm unsure of what is expected, I go and look at some huge project's NixOS module for inspiration, in this case [Mastodon](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-apps/mastodon.nix).
They just make the reverse proxy something you can turn off and customize if you like. So that way if you don't enable Nginx in this Cryptpad module, all it does is start the app server on localhost:3000. This leaves you free to configure a reverse proxy configuration of your choosing, including ACME settings.

So in my [latest commit](https://github.com/NixOS/nixpkgs/commit/73e025a03b84383b4e873ec68bf72bbbcf990c9d), yes some Nginx settings will get set, but you're free to opt out by setting `configureNginx` to false and do your own thing. That sidesteps the issue of what should and what shouldn't be enabled. What do you think?